runners/direct-java/src/test/java/org/apache/beam/runners/direct/portable/BundleFactoryOutputRecieverFactoryTest.java

```diff
@@ -0,0 +1,259 @@
+                    PaneInfo.NO_FIRING)));
+    secondReceiver.accept(secondElem);
+
+    Collection<WindowedValue<?>> outputs = new ArrayList<>();
+    for (UncommittedBundle<?> uncommitted : outputBundles) {
+      assertThat(uncommitted.getPCollection(), equalTo(fooPC));
+      Iterable<? extends WindowedValue<?>> elements =
+          uncommitted.commit(Instant.now()).getElements();
+      Iterables.addAll(outputs, elements);
+      assertThat(Iterables.size(elements), equalTo(1));
+    }
+    assertThat(outputs, containsInAnyOrder(firstElem, secondElem));
+  }
+
+  @Test
+  public void differentPCollectionsIndependent() throws Exception {
+    FnDataReceiver<WindowedValue<byte[]>> fooReceiver = factory.create(fooPC.getId());
+    MessageWithComponents fooSdkWireCoder =
+        WireCoders.createSdkWireCoder(fooPC, components, components::containsCoders);
+    Coder<WindowedValue<String>> fooSdkCoder =
+        (Coder<WindowedValue<String>>)
+            CoderTranslation.fromProto(
+                fooSdkWireCoder.getCoder(),
+                RehydratedComponents.forComponents(fooSdkWireCoder.getComponents()));
+    Coder<WindowedValue<byte[]>> fooRunnerCoder =
+        WireCoders.instantiateRunnerWireCoder(fooPC, components);
+
+    FnDataReceiver<WindowedValue<byte[]>> barReceiver = factory.create(barPC.getId());
+    MessageWithComponents barSdkWireCoder =
+        WireCoders.createSdkWireCoder(barPC, components, components::containsCoders);
+    Coder<WindowedValue<Integer>> barSdkCoder =
+        (Coder<WindowedValue<Integer>>)
+            CoderTranslation.fromProto(
+                barSdkWireCoder.getCoder(),
+                RehydratedComponents.forComponents(barSdkWireCoder.getComponents()));
+    Coder<WindowedValue<byte[]>> barRunnerCoder =
+        WireCoders.instantiateRunnerWireCoder(barPC, components);
+
+    WindowedValue<byte[]> fooElem =
+        CoderUtils.decodeFromByteArray(
+            fooRunnerCoder,
+            CoderUtils.encodeToByteArray(
+                fooSdkCoder,
+                WindowedValue.of(
+                    "1",
+                    new Instant(120),
+                    new IntervalWindow(new Instant(0), Duration.standardMinutes(5)),
+                    PaneInfo.NO_FIRING)));
+    fooReceiver.accept(fooElem);
+
+    WindowedValue<byte[]> barElem =
+        CoderUtils.decodeFromByteArray(
+            barRunnerCoder,
+            CoderUtils.encodeToByteArray(
+                barSdkCoder, WindowedValue.timestampedValueInGlobalWindow(2, new Instant(240))));
+    barReceiver.accept(barElem);
+
+    Collection<? super WindowedValue<?>> outputs = new ArrayList<>();
+    for (UncommittedBundle<?> uncommitted : outputBundles) {
+      WindowedValue<?> output = getOnlyElement(uncommitted.commit(Instant.now()).getElements());
+      if (fooPC.equals(uncommitted.getPCollection())) {
+        assertThat(output, equalTo(fooElem));
+      } else {
+        assertThat(
```

Comment:
This is much cleaner as its own else block as a fail(...), because then the fooPC and barPC output checks are mirrors of each other.